### PR TITLE
Add bitmanip test cases

### PR DESCRIPTION
**Pull request by rogerchang23424**
The test cases contains zba, zbb, zbc, zbs instructions

### CODE REVIEW

File: isa/rv32ub/Makefrag [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#=======================================================================
+# Makefrag for rv64ub tests
+#-----------------------------------------------------------------------
+
+rv32ub_sc_tests = \
+	andn \
+	bclr bclri \
+	bext bexti \
+  binv binvi \
```
Comment (aswaterman) [inline isa/rv32ub/Makefrag:9]: tabs in this file, not spaces

**Review by aswaterman — COMMENTED** [review]
Can you separate the tests into directories per extension, rather than lumping them together into "b"?

Can you tell us how you tested these?  Unfortunately, I have an older version of RISC-V binutils on this machine, so I can't compile and run the tests on Spike myself.

File: isa/rv32ub/Makefrag (hunk 1 above)
Comment (rogerchang23424) [reply isa/rv32ub/Makefrag]: Done

**Comment by rogerchang23424** [comment]
> Can you separate the tests into directories per extension, rather than lumping them together into "b"?

Done

> Can you tell us how you tested these? Unfortunately, I have an older version of RISC-V binutils on this machine, so I can't compile and run the tests on Spike myself.

The test cases require RISC-V gcc 12+ and binutils 2.38.0+ to run them, and make sure the version of Spike is latest.

https://godbolt.org/z/ehaE8Whb4

**Review by aswaterman — APPROVED** [review]

**Comment by aswaterman** [comment]
Thank you for this helpful contribution!

**Pull request merged**